viewer/src/main/webapp/viewer-html/components/EditBulk.js

```diff
@@ -0,0 +1,396 @@
+            var attributeName = input.name;
+            me.selectedFeatures.changeValue(attributeName, newValue);
+            me.makeFormFieldScalar(attributeName);
+        },
+        getListOfAttributeNamesThatAreMappedToFormFields: function () {
+            var me = this;
+            var mapped = [];
+            me.appLayer.attributes.forEach(function(attribute) {
+                var input = me.inputContainer.getForm().findField(attribute.name);
+                if (input !== null) {
+                    mapped.push(attribute.name);
+                }
+            });
+            return mapped;
+        }
+    });
+
+    function SelectedFeatures(mapComponent, attributeNames) {
+        var thisClass = this;
+        this.featureItems = [];
+        this.mapComponent = mapComponent;
+        this.changedValues = {};
+        this.attributeNames = attributeNames;
+
+        this.select = function (feature, clickCoord) {
+            if (!thisClass.isSelected(feature)) {
+                var item = new Item(feature, clickCoord);
+                thisClass.featureItems.push(item);
+                thisClass.showMarker(feature);
+            }
+        };
+
+        this.deselect = function (feature) {
+            if (thisClass.isSelected(feature)) {
+                thisClass.featureItems.splice(getIndex(feature), 1);
+                thisClass.hideMarker(feature);
+
+                if (thisClass.zeroSelected()) {
+                    this.changedValues = {}
+                }
+            }
+        };
+
+        this.deselectAll = function () {
+            var featureItemsCopy = thisClass.featureItems.map( function (item) { return item; });
+            for (var i = 0; i < featureItemsCopy.length; i++) {
+                thisClass.deselect(featureItemsCopy[i].feature);
+            }
+        };
+
+        this.changeValue = function (attributeName, newValue) {
+            thisClass.changedValues[attributeName] = newValue;
+        };
+
+        this.isSelected = function (feature) {
+            return -1 !== getIndex(feature);
+        };
+
+        this.numSelected = function () {
+            return thisClass.featureItems.length;
+        };
+
+        this.zeroSelected = function() {
+            return 0 === thisClass.featureItems.length;
+        };
+
+        this.showMarker = function (feature) {
+            var markerId = getMarkerId(feature);
+            var item = getItemForFeature(feature);
+            thisClass.mapComponent.setMarker(markerId, item.clickCoord.x, item.clickCoord.y, 'default');
+        };
+
+        this.hideMarker = function (feature) {
+            var markerId = getMarkerId(feature);
+            thisClass.mapComponent.removeMarker(markerId);
+        };
+
+        this.getSelectedFeatures = function () {
+            return thisClass.featureItems.map(function (item) { return shallowCloneObject(item.feature); });
+        };
+
+        function shallowCloneObject(original) {
```

Comment:
Don't know for sure, but doesn't ext have some functionality for this?

Reply:
not that i could find